subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -32,25 +32,85 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
         }
     }
 
+    def "detects changes when no files are in the project"() {
+        given:
+        def markerFile = file("input/marker")
+        buildFile << """
+            task build {
+              def inputFile = file("input/marker")
+              inputs.files inputFile
+              outputs.files "build/marker"
+              doLast {
+                println "exists: " + inputFile.exists()
+              }
+            }
+        """
+
+        when:
+        succeeds("build")
+        then:
+        output.contains "exists: false"
+
+        when:
+        waitBeforeModification(markerFile)
+        markerFile.text = "created"
+        then:
+        succeeds()
+        output.contains "exists: true"
+    }
+
+    def "detects changes when no snapshotting happens (#description)"() {
+        given:
+        def markerFile = file("input/marker")
+        buildFile << """
+            task build {
+              def inputFile = file("input/marker")
+              inputs.files inputFile
+              doLast {
+                println "value: " + inputFile.text
+              }
+            }
+        """
+
+        when:
+        markerFile.text = "original"
+        succeeds("build")
+        then:
+        output.contains "value: original"
+
+        when:
+        waitBeforeModification(markerFile)
+        markerFile.text = "changed"
+        then:
+        succeeds()
+        output.contains "value: changed"
+
+        where:
+        description      | taskConfiguration
+        "no outputs"     | ""
+        "untracked task" | "outputs.file('output/marker'); doNotTrackState('for test')"
+    }
+
     @ToBeFixedForConfigurationCache
     def "basic smoke test"() {
         given:
-        def markerFile = file("marker")
+        def markerFile = file("input/marker")
 
         when:
         markerFile.text = "original"
 
         buildFile << """
-            task echo {
-              inputs.files "marker"
+            task build {
+              inputs.files "input/marker"
+              outputs.files "build/marker"
               doLast {
-                println "value: " + file("marker").text
+                println "value: " + file("input/marker").text
               }
             }
         """
 
         then:
-        succeeds("echo")
+        succeeds("build")
```

Comment:
I'm a bit uneasy about the name `build` for this task. My first assumption reading the expectations in some other test was that this is the `build` lifecycle task added by the `base` plugin. Can we find a different name for this?

Reply:
I didn't like `echo`, since a task for triggering continuous build shouldn't only echo something. I see your point about the confusion with the lifecycle task.

How about `taskWithInputs` or `myTask`?

Reply:
Either sounds good, I just don't like reusing `build`.

Reply:
I went with `myTask`.